src/views/RegisterPage.vue

```diff
@@ -0,0 +1,281 @@
+});
+const subtitle = computed(() => {
+  return steps.value === 1 ? "Care to register?" : "Now your name";
+});
+
+const formData = ref<RegisterCredentials>({
+  email: "",
+  password: "",
+  first_name: "",
+  last_name: "",
+  date_of_birth: "",
+});
+const privacy_policy = ref(false);
+
+const formBlur = ref({
+  email: false,
+  password: false,
+  first_name: false,
+  last_name: false,
+  date_of_birth: false,
+  privacy_policy: false,
+});
+
+const emailValid = computed(() => {
+  return validateEmail(formData.value.email);
+});
+const passwordValidLength = computed(() => {
+  return formData.value.password.length > 7;
+});
+const passwordValidLetters = computed(() => {
+  return /[a-zA-Z]/.test(formData.value.password);
+});
+const passwordValidDigits = computed(() => {
+  return /\d/.test(formData.value.password);
+});
+const passwordValid = computed(() => {
+  return (
+    passwordValidLength.value &&
+    passwordValidLetters.value &&
+    passwordValidDigits.value
+  );
+});
+const firstNameValid = computed(() => {
+  return !!formData.value.first_name;
+});
+const lastNameValid = computed(() => {
+  return !!formData.value.last_name;
+});
+const dateOfBirthValid = computed(() => {
+  return validateDateOfBirth(formData.value.date_of_birth);
+});
+
+const emailValidation = computed(() => {
+  return !emailValid.value && formBlur.value.email
+    ? "Please provide a valid email address"
+    : "";
+});
+const firstNameValidation = computed(() => {
+  return !firstNameValid.value && formBlur.value.first_name
+    ? "Please provide a valid first name"
+    : "";
+});
+const lastNameValidation = computed(() => {
+  return !lastNameValid.value && formBlur.value.last_name
+    ? "Please provide a valid last name"
+    : "";
+});
+const dateOfBirthValidation = computed(() => {
+  return !dateOfBirthValid.value && formBlur.value.date_of_birth
+    ? "Please provide a valid date of birth"
+    : "";
+});
+const privacyPolicyValidation = computed(() => {
+  return !privacy_policy.value && formBlur.value.privacy_policy
+    ? "Please provide a valid date of birth"
+    : "";
+});
+
+const getWarningMessageClass = (validator: ComputedRef<string> | boolean) => {
+  if (formBlur.value.password)
+    return [
+      "register__warning",
+      validator ? "register__warning--green" : "register__warning--red",
+    ];
+};
+
+const submitForm = async () => {
+  if (steps.value === 1) {
+    formBlur.value.email = true;
+    formBlur.value.password = true;
+    if (!emailValid.value || !passwordValid.value) return;
+    steps.value += steps.value;
+  } else {
+    formBlur.value.first_name = true;
+    formBlur.value.last_name = true;
+    formBlur.value.date_of_birth = true;
+    formBlur.value.privacy_policy = true;
+    if (
+      !firstNameValid.value ||
+      !lastNameValid.value ||
+      !dateOfBirthValid.value ||
+      !privacy_policy.value
+    )
+      return;
+    try {
+      console.log(formData.value);
```

Comment:
Probably some sort of leftover that you could get rid of ;)

Reply:
correct :)